editor/editor_inspector.cpp

```diff
@@ -600,6 +600,40 @@ bool EditorProperty::is_selected() const {
 	return selected;
 }
 
+void EditorProperty::disable() {
```

Comment:
This function looks a little bit dirty. Most buttons (and hopefully spinboxes) have a disabled property, it might make sense to use that instead of using a mouse filter and changing focus mode.

Reply:
Currently the inspector uses `EditorSpinSlider`s which don't have a way to be disabled (I tried setting them as read-only, but could still edit them if a got focus through tabbing). I'll use the same approach used in `SpinBox` and add `set_editable` and `is_editable` to `EditorSpinSlider` (if it's considered a good approach).

Here's the code used when trying to use the read-only approach:
```cpp
void EditorProperty::disable() {
	// Make it visually apparent that the property is disabled
	Color color = get_modulate();
	color.a = color.a * 0.5;
	set_modulate(color);
	// Make it non editable
	set_mouse_filter(MOUSE_FILTER_IGNORE);
	set_focus_mode(FOCUS_NONE);
	int child_count = get_child_count();
	for (int i = 0; i < child_count; i++) {
		Node *child = get_child(i);
		Container *container = Object::cast_to<Container>(child);
		if (container) {
			container->set_mouse_filter(MOUSE_FILTER_IGNORE);
			container->set_focus_mode(FOCUS_NONE);
			int container_child_count = container->get_child_count();
			for (int j = 0; j < container_child_count; j++) {
				Node *container_child = container->get_child(j);
				EditorSpinSlider *spin_slider = Object::cast_to<EditorSpinSlider>(container_child);
				if (spin_slider) {
					spin_slider->set_read_only(true);
				}
			}
		} else {
			EditorSpinSlider *spin_slider = Object::cast_to<EditorSpinSlider>(child);
			if (spin_slider) {
				spin_slider->set_read_only(true);
			}
		}
	}
}
```

And here's a video to show the behavior:
![ImmenseCleverHypacrosaurus](https://user-images.githubusercontent.com/19125153/61365867-bfe53900-a888-11e9-84bb-9fa56dc93d66.gif)

Reply:
Is the function considered dirty in its current state?